builtin/providers/librato/resource_librato_metric_test.go

```diff
@@ -0,0 +1,230 @@
+					testAccCheckLibratoMetricName(&metric, name),
+					testAccCheckLibratoMetricType(&metric, []string{"gauge", "counter", "composite"}),
+					resource.TestCheckResourceAttr(
+						"librato_metric.foobar", "name", name),
+				),
+			},
+			{
+				PreConfig: sleep(t, 15),
+				Config:    gaugeMetricConfig(name, gauge, fmt.Sprintf("An updated test %s metric", gauge)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLibratoMetricExists("librato_metric.foobar", &metric),
+					testAccCheckLibratoMetricName(&metric, name),
+					testAccCheckLibratoMetricType(&metric, []string{"gauge", "counter", "composite"}),
+					resource.TestCheckResourceAttr(
+						"librato_metric.foobar", "name", name),
+				),
+			},
+		},
+	})
+}
+
+func TestAccLibratoCompositeMetric(t *testing.T) {
+	var metric librato.Metric
+	name := fmt.Sprintf("tftest-metric-%s", acctest.RandString(10))
+	composite := "composite"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLibratoMetricDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: compositeMetricConfig(name, composite, fmt.Sprintf("A test %s metric", composite)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLibratoMetricExists("librato_metric.foobar", &metric),
+					testAccCheckLibratoMetricName(&metric, name),
+					testAccCheckLibratoMetricType(&metric, []string{"gauge", "counter", "composite"}),
+					resource.TestCheckResourceAttr(
+						"librato_metric.foobar", "name", name),
+				),
+			},
+			{
+				PreConfig: sleep(t, 15),
+				Config:    compositeMetricConfig(name, composite, fmt.Sprintf("An updated test %s metric", composite)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLibratoMetricExists("librato_metric.foobar", &metric),
+					testAccCheckLibratoMetricName(&metric, name),
+					testAccCheckLibratoMetricType(&metric, []string{"gauge", "counter", "composite"}),
+					resource.TestCheckResourceAttr(
+						"librato_metric.foobar", "name", name),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckLibratoMetricDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*librato.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "librato_metric" {
+			continue
+		}
+
+		fmt.Printf("*** testAccCheckLibratoMetricDestroy ID: %s\n", rs.Primary.ID)
+		_, _, err := client.Metrics.Get(rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Metric still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckLibratoMetricName(metric *librato.Metric, name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if metric.Name == nil || *metric.Name != name {
+			return fmt.Errorf("Bad name: %s", *metric.Name)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckLibratoMetricType(metric *librato.Metric, validTypes []string) resource.TestCheckFunc {
```

Comment:
It looks like this could be `testAccCheckLibratoMetricType(metric *librato.Metric, wantType string) 
 resource.TestCheckFunc` and provide more valuable test coverage for the cases where it's used.